library/nametab.ml

```diff
@@ -550,33 +551,47 @@ let path_of_universe mp =
   UnivIdMap.find mp !the_univrevtab
 
 (* Shortest qualid functions **********************************************)
-
+(* Names.GlobRef.t -> Libnames.qualid *)
 let shortest_qualid_of_global ?loc ctx ref =
   let open GlobRef in
-  match ref with
+  if !Flags.fqn_print then
+    Libnames.qualid_of_path ?loc (path_of_global ref)
+  else
+    match ref with
     | VarRef id -> make_qualid ?loc DirPath.empty id
     | _ ->
-        let sp =  ExtRefMap.find (TrueGlobal ref) !the_globrevtab in
-        ExtRefTab.shortest_qualid ?loc ctx sp !the_ccitab
+       let sp =  ExtRefMap.find (TrueGlobal ref) !the_globrevtab in
+       ExtRefTab.shortest_qualid ?loc ctx sp !the_ccitab
 
 let shortest_qualid_of_abbreviation ?loc ctx kn =
-  let sp = path_of_abbreviation kn in
+  if !Flags.fqn_print then
+    Libnames.qualid_of_path ?loc (path_of_abbreviation kn)
+  else
+    let sp = path_of_abbreviation kn in
     ExtRefTab.shortest_qualid ?loc ctx sp !the_ccitab
 
 let shortest_qualid_of_module ?loc mp =
```

Comment:
I dont see a `path_of_module` in this file to fix this function to support `Set Printing FullyQualifiedNames.`

Reply:
your grep is broken then

Reply:
your repo is outdated, rebase on master